packages/@aws-cdk/aws-lambda/lib/layer-version.ts

```diff
@@ -0,0 +1,163 @@
+import cdk = require('@aws-cdk/cdk');
+import { Code, InlineCode } from './code';
+import { Runtime } from './runtime';
+
+export interface LayerVersionProps {
+  /**
+   * The runtimes that this layer is compatible with.
+   *
+   * @default All runtimes are supported
+   */
+  compatibleRuntimes?: Runtime[];
+
+  /**
+   * The content of this Layer. Using the ``InlineCode`` class is not permitted.
+   */
+  code: Code;
+
+  /**
+   * The description the this Lambda Layer.
+   */
+  description?: string;
+
+  /**
+   * The SPDX licence identifier or URL to the license file for this layer.
+   */
+  license?: string;
+
+  /**
+   * The name of the layer.
+   * @default a name will be generated.
+   */
+  name?: string;
+}
+
+/**
+ * A reference to a Lambda Layer version.
+ */
+export abstract class LayerVersionRef extends cdk.Construct {
+  /**
+   * Imports a Layer that has been defined externally.
+   *
+   * @param parent the parent Construct that will use the imported layer.
+   * @param id     the id of the imported layer in the construct tree.
+   * @param props  the properties of the imported layer.
+   */
+  public static import(parent: cdk.Construct, id: string, props: LayerVersionRefProps): LayerVersionRef {
+    return new ImportedLayerVersionRef(parent, id, props);
+  }
+
+  /**
+   * The ARN of the Lambda Layer version that this Layer defines.
+   */
+  public abstract readonly layerVersionArn: string;
+
+  /**
+   * The runtimes compatible with this Layer.
+   */
+  public abstract readonly compatibleRuntimes?: Runtime[];
+
+  /**
+   * Grants usage of this layer to specific entities. Usage within the same account where the layer is defined is always
+   * allowed and does not require calling this method. Note that the principal that creates the Lambda function using
+   * the layer (for example, a CloudFormation changeset execution role) also needs to have the
+   * ``lambda:GetLayerVersion`` permission on the layer version.
+   *
+   * @param awsAccountId   the AWS account ID that will be granted access to the layer, or '*' to grant usage to any/all
+   *                       AWS accounts. The default is '*'.
+   * @param organizationId when using '*' as the ``awsAccountId``, a organization ID can be procided to restrict usage
+   *                       to accounts that are memeber of a given organization.
+   */
+  public grantUsage(awsAccountId: string = '*', organizationId?: string): this {
```

Comment:
So what does “grantUsage()” (with no arguments) do?

I would use an opts hash (names arguments) in this case

Reply:
That would grant public access. I'm changing to using a hash argument - so this becomes a required argument (aka - you have to be explicit about it if you want it public).